docs/source/federated-types/sharing-types.mdx

```diff
@@ -0,0 +1,404 @@
+
+<CodeColumns>
+
+```graphql{2}
+# Subgraph A
+type Position @shareable {
+  x: Int!
+  y: Int!
+}
+```
+
+```graphql{2}
+# Subgraph B
+type Position @shareable {
+  x: Int!
+  y: Int!
+}
+```
+<></>
+
+</CodeColumns>
+
+> If a type or field is marked `@shareable` in _any_ subgraph, it must be marked `@shareable` or [`@external`](../federation-spec/#external) in _every_ subgraph that defines it. Otherwise, composition fails.
+
+The `@shareable` directive enables multiple subgraphs to resolve shared fields. So in the example above, both subgraphs A _and_ B can resolve the `Position.x` and `Position.y` fields.
+
+> ⚠️ **Important:** If multiple subgraphs can resolve a field, **make sure each subgraph's resolver for that field behaves identically.** Otherwise, queries might return inconsistent results depending on which subgraph resolves the field.
+
+Adding the `@shareable` directive to a type is a shortcut to sharing every field within that type. You can also share individual fields, like so:
+
+<CodeColumns>
+
+```graphql{3}
+# Subgraph A
+type Food {
+  name: String! @sharable
+  cost: Int!
+}
+```
+
+```graphql{3}
+# Subgraph B
+type Food {
+  name: String!  @sharable
+  inStock: Boolean!
+}
+```
+
+<></>
+
+</CodeColumns>
+
+If you'd like to learn how shared fields can differ across subgraphs, see [Varying shared fields](#varying-shared-fields).
+
+## Explicit and implicit type sharing
+
+### Objects
+
+By default, in Federation 2 schemas, Object type **fields** are not shared. This means that you cannot redefine an existing Object type's field without explicitly sharing that field.
+
+For example, let's say you'd like to define a `Product` type in two subgraphs:
+
+<p style="margin-bottom: 0">❌</p>
+
+<CodeColumns>
+
+```graphql{3, 4}
+# Subgraph A
+type Product {
+  name: String!
+  price: Int
+}
+```
+
+```graphql{3, 4}
+# Subgraph B
+type Product {
+  name: String!
+  price: Int
+}
+```
+
+</CodeColumns>
+
+Attempting to compose these two subgraph schemas together will [**break composition**](./composition/#breaking-composition). The gateway doesn't know _which subgraph_ is responsible for resolving the `Product` type's fields.
+
+If we'd like both Subgraph A _and_ Subgraph B to be able to resolve the fields for `Product`, we can [explicitly share](#using-shareable) that type:
+
+<p style="margin-bottom: 0">✅</p>
+
+<CodeColumns>
+
+```graphql{2}
+# Subgraph A
+type Product @shareable {
+  name: String!
+  price: Int
+}
+```
+
+```graphql{2}
+# Subgraph A
+type Product @shareable {
+  name: String!
+  price: Int
+}
+```
+
+</CodeColumns>
+
+Both subgraphs A and B can now resolve the `name` and `price` fields for the `Product` type, and our subgraph schema will successfully compose into a supergraph schema.
+
+As an alternate course of action, if you wanted Subgraph A and B to resolve their own fields, you could instead define `Product` as an [entity](../entities). 
```

Comment:
"As an alternative, if you want Subgraphs A and B to resolve _different_ fields of `Product`, you can designate the `Product` type as an entity."

Also this line might look nice in a blockquote